### DISCUSSION TITLE
Huggingface setup

### DISCUSSION THREAD

**Discussion by lucadanu** [nhaouari/obsidian-textgenerator-plugin#64] [Q&A] [answered]
Hi, I don't understand well how to setup the Huggingface template.
Once I have created the API key where is the templates file to insert it?

**Comment by nhaouari** [comment, marked as answer]
Hi, by default, you find the  Huggingface templates here: 
![image](https://user-images.githubusercontent.com/9850722/208674274-d346d3b5-d114-4211-b12a-eac4e3fd7e3b.png)

**Reply by lucadanu** [reply]
Thank you very much!

**Reply by nhaouari** [reply]
You're welcome

**Reply by lucadanu** [reply]
I pasted the APY key inside each huggingface templates but they don't work properly. For example when I put the text inside the box in order to be summarized, i just see the loading process but nothing is created

**Reply by nhaouari** [reply]
Could you please try opening the Developer Tools console (Ctrl+Shift+i) and seeing if there are any errors being displayed there?

**Reply by lucadanu** [reply]
plugin: obsidian-textgenerator-plugin:62 Uncaught (in promise) Error: Request failed, status 400
    at new t (app.js:1:710073)
    at Aw (app.js:1:710265)
    at app.js:1:710942
    at app.js:1:235505
    at Object.next (app.js:1:235610)
    at a (app.js:1:234349)

This is the error is showing up

**Reply by lucadanu** [reply]
@nhaouari Could you please help me to figure out what is going wrong with hugging face models in my case? I cannot use them in any prompts

**Reply by nhaouari** [reply]
I couldn't reproduce the error. It works fine for me.

**Reply by lucadanu** [reply]
I see..I guess I missed a step on setting up.
I install, enable the plugin. Install the 3 different templates packages on Obsidian. 
I create the Key token on huggingface as a read key, I paste it as field in the "Authorization" properties inside prompts files of Huggingface templates.

Then I choose the command "Show model from Template",  I copy and paste a paragraph on the window to try Summarize but nothing happen.
Have I missed something?

**Reply by nhaouari** [reply]
> I see..I guess I missed a step on setting up. I install, enable the plugin. Install the 3 different templates packages on Obsidian. I create the Key token on huggingface as a read key, I paste it as field in the "Authorization" properties inside prompts files of Huggingface templates.
> 
> Then I choose the command "Show model from Template", I copy and paste a paragraph on the window to try Summarize but nothing happen. Have I missed something?


Could you please update the huggingface template and reintroduce the api key.
